### PR TITLE
Bugfix Summarization Score

### PR DESCRIPTION
**Pull request by emreds**
Hi! 

I was experimenting with Summarization Score and found an issue that when there are no questions generated by the LLM, it throws division by zero error due to 0 generated answers. 

## Notable Changes
- Added an assertion to keyphrases, questions and answers generation functions.
- Changed the llm response variable name `answer` to `response` to prevent possible confusion with score related `answer`.

Please let me know if there is something I can add. 

Cheers!

### CODE REVIEW

**Review by jjmachan — COMMENTED** [review]
@emreds thanks a lot for pushing this PR and helping us Improve Ragas. I was however not sure if an assertion here is the best flow.

it will break the user code right if the assertions fail? Ideally the behaviour we want is to use `logger.error()` to print out an explainable error and then return a Nan value (or throw and exception since executor should catch it). 

what do you think?

**Comment by emreds** [comment]
@jjmachan sure, makes more sense. I will update it.

File: src/ragas/metrics/_summarization.py [hunk 1]
```diff
@@ -232,7 +235,9 @@ async def _get_questions(
             result_text, p_value, self.llm, self.max_retries
         )
         
-        assert response and response.questions, "No questions generated, unable to calculate the score."
+        if not response or not response.questions:
+            logging.error("No questions generated, unable to calculate the score.")
+            return math.nan
```
Comment (jjmachan) [inline src/ragas/metrics/_summarization.py:240]:
```suggestion
            return np.nan
```

**Review by jjmachan — CHANGES_REQUESTED** [review]
math.nan to np.nan. 

I've changed one, do change the others too?

just to keep things consistent throughout the codebase 🙂

File: src/ragas/metrics/_summarization.py (hunk 1 above)
Comment (emreds) [reply src/ragas/metrics/_summarization.py]: Sure, I have changed the all and made a new commit.

**Review by jjmachan — APPROVED** [review]
sweet! thanks a lot @emreds ❤️

**Comment by jjmachan** [comment]
@emreds I made some small changes for fixing the type error but now it still returns an empty list. This is so that the types make sense. but I'm not sure if it logically correct

**Pull request merged**